src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GeyserOptions.kt

```diff
@@ -0,0 +1,68 @@
+package at.hannibal2.skyhanni.features.fishing.trophy
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.FishingBobberCastEvent
+import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
+import at.hannibal2.skyhanni.events.ReceiveParticleEvent
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceTo
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RenderUtils.drawFilledBoundingBox_nea
+import at.hannibal2.skyhanni.utils.SpecialColour
+import net.minecraft.entity.projectile.EntityFishHook
+import net.minecraft.util.AxisAlignedBB
+import net.minecraft.util.EnumParticleTypes
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+
+class GeyserOptions {
+    private var bobber: EntityFishHook? = null
+    private var geyser: LorenzVec? = null
+
+    @SubscribeEvent(priority = EventPriority.LOW, receiveCanceled = true)
+    fun onPacket(event: ReceiveParticleEvent) {
+        if (!shouldProcessParticles()) return
+
+        if (event.type == EnumParticleTypes.CLOUD) {
+            geyser = event.location
+        }
+
+        if (isHideParticlesEnabled() && geyser != null && bobber != null && event.type == EnumParticleTypes.CLOUD) {
+            hideGeyserParticles(event)
+        }
+    }
+
+    @SubscribeEvent
+    fun onBobberThrow(event: FishingBobberCastEvent) {
+        bobber = event.bobber
+    }
+
+    @SubscribeEvent
+    fun onRenderWorld(event: LorenzRenderWorldEvent) {
+        if (shouldDrawBoundingBox()) {
+            val geyserBox = AxisAlignedBB(
+                geyser!!.x - 1, 118.0 - 0.1, geyser!!.z - 1,
+                geyser!!.x + 1, 118.0 - 0.09, geyser!!.z + 1
+            )
```

Comment:
would do something like this to follow the [contributing guide](https://github.com/hannibal002/SkyHanni/blob/d9ffe4a485f58105bdc93ba572ed4e541224c6c8/CONTRIBUTING.md?plain=1#L73)
```kotlin
val newGeyser = geyser ?: return
val geyserBox = AxisAlignedBB(
    newGeyser.x - 1, 118.0 - 0.1, newGeyser.z - 1,
    newGeyser.x + 1, 118.0 - 0.09, newGeyser.z + 1
)
```

Reply:
done 👍